### PR TITLE
bpo-40956: Convert _sqlite3.Connection to Argument Clinic, part 1 of 2

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue40956

### CODE REVIEW

**Comment by erlend-aasland** [comment]
@corona10: As proposed on #23057, I've split this up in two parts. This is the first, and larger (by line count) part, however probably the easiest to review, as the AC conversion is straight-forward.

**Comment by erlend-aasland** [comment]
@corona10 Have you had time to look at this? Let me know if still think it's too large for a single review.

**Comment by corona10** [comment]
@erlend-aasland I will take a look ;)

**Comment by erlend-aasland** [comment]
@corona10 Thank you :) 🙏🏻

File: Modules/_sqlite/connection.c [hunk 1]
```diff
@@ -982,22 +1017,26 @@ static void _trace_callback(void* user_arg, const char* statement_string)
 #endif
 }
 
-static PyObject* pysqlite_connection_set_authorizer(pysqlite_Connection* self, PyObject* args, PyObject* kwargs)
-{
-    PyObject* authorizer_cb;
+/*[clinic input]
+_sqlite3.Connection.set_authorizer as pysqlite_connection_set_authorizer
+
+    authorizer_callback as authorizer_cb: object
```
Comment (corona10) [inline Modules/_sqlite/connection.c:1023]: Should take keyword arguments also.

**Review by corona10 — CHANGES_REQUESTED** [review]

File: Modules/_sqlite/connection.c (hunk 1 above)
Comment (erlend-aasland) [reply Modules/_sqlite/connection.c:1023]: PTAL :)

**Comment by erlend-aasland** [comment]
I have made the requested changes; please review again

File: Modules/_sqlite/connection.c [hunk 2]
```diff
@@ -1038,23 +1079,23 @@ static PyObject* pysqlite_connection_set_progress_handler(pysqlite_Connection* s
     Py_RETURN_NONE;
 }
 
-/*
```
Comment (corona10) [inline Modules/_sqlite/connection.c:1041]: Can we leave this comment somewhere?

**Review by corona10 — APPROVED** [review]
LGTM except for nit comment ;)

File: Modules/_sqlite/connection.c (hunk 2 above)
Comment (erlend-aasland) [reply Modules/_sqlite/connection.c:1041]: Yes, good catch!

Comment (erlend-aasland) [reply Modules/_sqlite/connection.c:1041]: How about dc1fd83dd6 ?

**Comment by erlend-aasland** [comment]
> LGTM except for nit comment ;)

I appreciate a thorough review! :) PTAL

File: Modules/_sqlite/connection.c [hunk 3]
```diff
@@ -1038,28 +1079,34 @@ static PyObject* pysqlite_connection_set_progress_handler(pysqlite_Connection* s
...
- */
-static PyObject* pysqlite_connection_set_trace_callback(pysqlite_Connection* self, PyObject* args, PyObject* kwargs)
-{
-    PyObject* trace_callback;
+/*[clinic input]
+_sqlite3.Connection.set_trace_callback as pysqlite_connection_set_trace_callback
 
-    static char *kwlist[] = { "trace_callback", NULL };
+    trace_callback: object
+    /
```
Comment (serhiy-storchaka) [inline Modules/_sqlite/connection.c:1086]: Should take keyword arguments also.

File: Modules/_sqlite/connection.c [hunk 4]
```diff
@@ -1080,19 +1127,26 @@ static PyObject* pysqlite_connection_set_trace_callback(pysqlite_Connection* sel
 }
 
 #ifndef SQLITE_OMIT_LOAD_EXTENSION
-static PyObject* pysqlite_enable_load_extension(pysqlite_Connection* self, PyObject* args)
+/*[clinic input]
+_sqlite3.Connection.enable_load_extension as pysqlite_connection_enable_load_extension
+
+    enable as onoff: int
```
Comment (serhiy-storchaka) [inline Modules/_sqlite/connection.c:1133]: `bool(accept={int})`

Semantically it is a bool.

File: Modules/_sqlite/connection.c (hunk 3 above)
Comment (erlend-aasland) [reply Modules/_sqlite/connection.c:1086]: Fixed by @corona10 in #23837

File: Modules/_sqlite/connection.c (hunk 4 above)
Comment (erlend-aasland) [reply Modules/_sqlite/connection.c:1133]: > `bool(accept={int})`

Nice trick! Thanks

Fixed by @corona10 in #23837

**Pull request merged**